newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,221 +65,192 @@ type State = {|
-            onInputValueChange={inputValue => {
-              this.setState({ inputValue });
-            }}
-            itemToString={item =>
-              item ? (item.type === 'separator' ? '' : item.value) : ''
-            }
-          >
-            {({
-              getInputProps,
-              getItemProps,
-              getLabelProps,
-              getMenuProps,
-
-              // Actions:
-              closeMenu,
-              openMenu,
-
-              // State:
-              highlightedIndex,
-              inputValue,
-              isOpen,
-              selectedItem,
-            }) => {
-              const { onBlur, ...inputProps } = getInputProps({
-                placeholder:
-                  typeof props.hintText === 'string'
-                    ? props.hintText
-                    : i18n._(props.hintText),
-                disabled: props.disabled,
-              });
-
-              // Wrap onBlur to close the menu and commit the changes to the value
-              const wrappedOnBlur = event => {
-                onBlur(event);
-
-                // Downshift will, after the blur event, reset the state so that the
-                // input value is itemToString applied to the selected item - which can
-                // be null as we allow whatever value is entered, even an incomplete one.
-                // Use a setTimeout to clear the inputValue just after Downshift has
-                // changed inputValue.
-                // (this is purely "visual", the onChange props is properly called, this
-                // being done or not)
-                setTimeout(() => {
-                  this.setState({
-                    inputValue: null,
-                  });
-                });
-
-                // Also close the menu
-                closeMenu();
-
-                // Call onChange with whatever value is entered, even an incomplete one.
-                props.onChange(event.currentTarget.value);
-
-                if (props.onBlur) props.onBlur(event);
-              };
-
-              const onFocus = event => {
-                if (props.openOnFocus) openMenu();
-                this.setState({
-                  inputValue: props.value,
-                });
-              };
-
-              return (
-                <div
-                  style={{
-                    ...styles.container,
-                    flexGrow: props.fullWidth ? 1 : undefined,
-                  }}
-                >
-                  {renderTextField({
-                    disabled: props.disabled,
-                    label: props.floatingLabelText,
-                    id: props.id,
-
-                    // Error handling:
-                    error: !!props.errorText,
-                    helperText: props.errorText || helperText,
-
-                    // Display:
-                    InputLabelProps: getLabelProps({ shrink: true }),
+    const handleOptionLabel = (option: DataSource | null): string => {
```

Comment:
Careful about your functions naming :) This is not "handling" anything, just returning to you the label to display.
```suggestion
    const getOptionLabel = (option: DataSource | null): string => {
```